### ISSUE TITLE
ThrowingExceptionInMain rule should consider main() function with no parameters

### ISSUE THREAD

**Issue by schalkms** [arturbosch/detekt#2248]
As of Kotlin 1.3, the explicit Array<String> can be omitted but will still be available in the byte code.
[Reference](https://stackoverflow.com/a/54325625)

## Expected Behavior
The rule should also flag parameterless `main()` functions that throw an exception.

```kotlin
fun main() { throw IllegalArgumentException() }
```

## Observed Behavior
The rule ignores parameterless `main()` functions.

## Your Environment

* Version of detekt used: 1.3.1
* Operating System and version: macOS 10.14.6
